libethereum/EthereumPeer.cpp

```diff
@@ -236,7 +236,10 @@ bool EthereumPeer::isCriticalSyncing() const
 
 bool EthereumPeer::interpret(unsigned _id, RLP const& _r)
 {
-	assert(m_observer);
+	auto observer = m_observer.lock();
+	assert(observer);
```

Comment:
I think we should better return false here instead of assert, for the case when network message handler is called at the time observer doesn't exist anymore

Reply:
OK.